racun-service/src/main/java/rs/edu/raf/banka/racun/controller/UgovorController.java

```diff
@@ -0,0 +1,163 @@
+        return ResponseEntity.ok(ugovori);
+    }
+
+    @GetMapping(value = "/get/draft/", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getAllDraft(@RequestHeader("Authorization") String token) {
+        var ugovori = ugovorService.getAllDraft();
+        return ResponseEntity.ok(ugovori);
+    }
+
+    @GetMapping(value = "/get/company/{kompanijaId}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getAllKompanija(@RequestHeader("Authorization") String token, @PathVariable Long kompanijaId) {
+        try {
+            List<Ugovor> ugovori = ugovorService.getAllByCompany(kompanijaId);
+            return ResponseEntity.ok(ugovori);
+
+        } catch (Exception ex) {
+            return ResponseEntity.badRequest().body(ex.getMessage());
+        }
+    }
+
+    @GetMapping(value = "/get/company/{kompanijaId}/finalized", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getAllKompanijaFinalized(@RequestHeader("Authorization") String token, @PathVariable Long kompanijaId) {
+        try
+        {
+            var ugovori = ugovorService.getAllByCompanyFinalized(kompanijaId);
+            return ResponseEntity.ok(ugovori);
+        }
+        catch (Exception ex)
+        {
+            return ResponseEntity.badRequest().body(ex.getMessage());
+        }
+    }
+
+    @GetMapping(value = "/get/company/{kompanijaId}/draft", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getAllKompanijaDraft(@RequestHeader("Authorization") String token, @PathVariable Long kompanijaId) {
+        try
+        {
+            var ugovori = ugovorService.getAllByCompanyDraft(kompanijaId);
+            return ResponseEntity.ok(ugovori);
+        }
+        catch (Exception ex)
+        {
+            return ResponseEntity.badRequest().body(ex.getMessage());
+        }
+    }
+
+    @PostMapping(value = "/create", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> createUgovor(@RequestHeader("Authorization") String token, @RequestBody UgovorCreateRequest request) {
+        try
+        {
+            var ugovor = ugovorService.createUgovor(request);
+            return ResponseEntity.ok(ugovor);
+        }
+        catch (Exception ex)
+        {
+            return ResponseEntity.badRequest().body(ex.getMessage());
+        }
+    }
+
+    @PutMapping(value = "/modify", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> modifyUgovor(@RequestHeader("Authorization") String token, @RequestBody UgovorUpdateRequest request) {
+        try
+        {
+            var result = ugovorService.modifyUgovor(request);
+            return ResponseEntity.ok(result);
+        }
+        catch (Exception ex)
+        {
+            return ResponseEntity.badRequest().body(ex.getMessage());
+        }
+    }
+
+    @PutMapping(value = "/modify/{id}/addDocument", produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
Neka ovo bude finalize ruta, tj. da automatski radi finalizaciju. Koliko vidim, to je već negde cilj, ali samo promeni rutu u nešto poput `/finalize/{id}`. Takođe, stavi da je ovo POST zahtev.